check-project-version/action.yml

```diff
@@ -0,0 +1,50 @@
+name: Check project version
+description: Check the version of a Python project against an expected version, such as a release tag.
+inputs:
+  project-directory:
+    description: Path to the directory containing pyproject.toml.
+    default: ${{ github.workspace }}
+  expected-version:
+    description: >
+      The expected version. By default, this is `github.ref_name`, which is the
+      tag for a `release` event. If the version has a leading 'v', it will be
+      stripped.
+    default: ${{ github.ref_name }}
+runs:
+  using: composite
+  steps:
+  - run: echo "::notice title=Notice::Test"
+    shell: bash
+  - name: Check project version
+    run: |
+      echo "::notice title=Notice::A"
+      project_version="$(poetry version --short)"
+      expected_version="${{ inputs.expected-version }}"
+      # Strip the leading 'v', in case this is a GitHub release tag.
+      expected_version="${expected_version#v}"
+      echo "::notice title=Notice::B"
+
+      error_message="$(cat <<EOF
```

Comment:
Using `read -r -d '' error_message <<EOF` fails because it sets a nonzero exit code